WordPress/src/main/java/org/wordpress/android/models/Note.kt

```diff
@@ -0,0 +1,525 @@
+/**
+ * Note represents a single WordPress.com notification
+ */
+package org.wordpress.android.models
+
+import android.text.Spannable
+import android.text.SpannableString
+import android.text.TextUtils
+import android.util.Base64
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+import org.wordpress.android.fluxc.model.CommentModel
+import org.wordpress.android.fluxc.model.CommentStatus
+import org.wordpress.android.ui.notifications.utils.NotificationsUtilsWrapper
+import org.wordpress.android.util.AppLog
+import org.wordpress.android.util.DateTimeUtils
+import org.wordpress.android.util.DateUtils.addDays
+import org.wordpress.android.util.DateUtils.addMonths
+import org.wordpress.android.util.DateUtils.addWeeks
+import org.wordpress.android.util.DateUtils.isSameDay
+import org.wordpress.android.util.JSONUtils
+import org.wordpress.android.util.StringUtils
+import java.io.UnsupportedEncodingException
+import java.util.Date
+import java.util.EnumSet
+import java.util.zip.DataFormatException
+import java.util.zip.Inflater
+
+class Note {
+    val id: String
+
+    private var mNoteJSON: JSONObject?
+    private var mActions: JSONObject? = null
+    private var mLocalStatus: String? = null
+
+    private val mSyncLock = Any()
```

Comment:
From what I could tell, we the current synchronized code doesn't require reentrancy. I wonder whether we could use a `Mutex` here instead? 🤔

Reply:
I experimented a bit with using `Mutex` but this would require introducing coroutines and changes outside the scope of the `Note` model. Tbh I'm not sure if locking is needed in all cases when accessing the json object 🤔